### PR TITLE
STAC-13204: Splunk topology, including integration tests

### PR DESCRIPTION
**Pull request by craffit**
### Step 2: Description of changes
- Add splunk_topology and port unit tests
- Add new integration tests for splunk
- Drop the polling interval in favor of min_collection_interval
- Add installing of base libraries in checkdev

### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test	

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: stackstate_checks_base/stackstate_checks/base/stubs/aggregator.py [hunk 1]
```diff
@@ -71,6 +71,7 @@ def service_checks(self, name):
         """
         Return the service checks received under the given name
         """
+        print(self._service_checks)
```
Comment (craffit) [inline stackstate_checks_base/stackstate_checks/base/stubs/aggregator.py:74]: remove this

File: splunk_topology/tests/test_unit.py [hunk 2]
```diff
@@ -0,0 +1,1117 @@
...
+                "name": "relations",
+                "parameters": {}
+            }],
+            'tags': ['mytag', 'mytag2']
+        }
+
+        init_config = {'default_polling_interval_seconds': 15}
+
+        self.check = MockedSplunkTopology(self.CHECK_NAME, init_config, {}, [instance])
+        print(self.check.init_config)
```
Comment (craffit) [inline splunk_topology/tests/test_unit.py:643]: not needed

File: splunk_topology/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# PostgreSQL Integration
```
Comment (vishalxebia) [inline splunk_topology/README.md:1]: should be `splunk topology`.

File: splunk_topology/tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+
+def connect_to_splunk():
+    SplunkClient(InstanceConfig(_empty_instance, {})).auth_session()
+
+
+@pytest.fixture(scope='session')
+def test_environment():
+    """
+    Start a standalone postgres server requiring authentication.
```
Comment (vishalxebia) [inline splunk_topology/tests/conftest.py:37]: typo `postgres`.

File: splunk_topology/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            conditions=[WaitFor(connect_to_splunk)],
+    ):
+        yield True
+
+
+# this fixture is used for checksdev env start
+@pytest.fixture(scope='session')
+def sts_environment(test_environment):
+    """
+    Start a standalone postgres server requiring authentication.
```
Comment (vishalxebia) [inline splunk_topology/tests/conftest.py:50]: same here `postgres` typo.

File: stackstate_checks_dev/stackstate_checks/dev/docker.py [hunk 6]
```diff
@@ -213,7 +213,7 @@ def __init__(self, compose_file, build=False, service_name=None):
             self.command.append(self.service_name)
 
     def __call__(self):
-        return run_command(self.command, check=True)
+        return run_command(self.command, "both", check=True)
```
Comment (vishalxebia) [inline stackstate_checks_dev/stackstate_checks/dev/docker.py:216]: what does `both` do here?

Comment (craffit) [reply stackstate_checks_dev/stackstate_checks/dev/docker.py:216]: it makes sure we get stdout and stderr from the docker compose

**Review by vishalxebia — APPROVED** [review]

**Pull request merged**